### PR TITLE
chore(doc): add stable selector to documentation

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

Web Components | Add Stable Selectors to Card Link #4356
Web Components | Add Stable Selectors to Horizontal Rule #4362
Web Components | Add Stable Selectors to Leadspace #4365
Web Components | Add Stable Selectors to Image #4364
Web Components | Add Stable Selectors to Quote #4368
Web Components | Add Stable Selectors/Prop Table to Callout Quote #4354

### Description

Add stable selector documentation to READMEs.

### CODE REVIEW

File: packages/web-components/src/components/leadspace/__stories__/README.stories.mdx [hunk 1]
```diff
@@ -45,4 +45,13 @@ import '@carbon/ibmdotcom-web-components/es/components/button-group/button-group
...
 
 <Props of="dds-leadspace" />
+
+## Stable selectors
+
+See [our README](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/master/packages/web-components#stable-selectors-for-analytics-and-integratione2e-testing-in-web-components) to see how Web Components selector and `data-autoid` should be used.
+
+| Web Components selector | Compatibility selector         | Description |
+| ----------------------- | ------------------------------ | ----------- |
+| `<dds-leadspace`        | `data-autoid="dds--leadspace"` | Component   |
```
Comment (asudoh) [inline packages/web-components/src/components/leadspace/__stories__/README.stories.mdx:55]:
```suggestion
| `<dds-leadspace>`       | `data-autoid="dds--leadspace"` | Component   |
```

**Review by asudoh — COMMENTED** [review]
Thank you for jumping in @annawen1 ! Do you want to add `stableSelector` static property to `horizontal-rule.ts` and `headspace.ts`?

**Comment by annawen1** [comment]
@asudoh I added documentation for a few more components, can you take a look? Thanks!

**Review by asudoh — APPROVED** [review]
Thanks @annawen1 for the updates!

File: packages/web-components/src/components/callout-quote/__stories__/README.stories.mdx [hunk 2]
```diff
@@ -39,4 +40,16 @@ import '@carbon/ibmdotcom-web-components/es/components/callout-quote/callout-quo
 </dds-callout-quote>
 ```
 
-<Description markdown={contributing} />
\ No newline at end of file
+## Props
+
+<Props of="dds-hr" />
```
Comment (asudoh) [inline packages/web-components/src/components/callout-quote/__stories__/README.stories.mdx:45]:
```suggestion
<Props of="dds-callout-quote" />
```

**Review by asudoh — COMMENTED** [review]
Thank you for covering additional things @annawen1!

**Comment by asudoh** [comment]
@annawen1 Just one more thing - Would you update the snapshot? Thanks!

**Pull request merged**